Client/BraveSkus/BraveSkusManager.swift

```diff
@@ -0,0 +1,146 @@
+  
+  func refreshOrder(for orderId: String, domain: String, resultJSON: @escaping (Any?) -> Void) {
+    sku.refreshOrder(domain, orderId: orderId) { completion in
+      do {
+        guard let data = completion.data(using: .utf8) else { return }
+        let json = try JSONSerialization.jsonObject(with: data, options: .fragmentsAllowed)
+        Logger.module.debug("refreshOrder json parsed successfully")
+        resultJSON(json)
+      } catch {
+        resultJSON(nil)
+        Logger.module.error("refrshOrder: Failed to decode json: \(error.localizedDescription)")
+      }
+    }
+  }
+  
+  func fetchOrderCredentials(for orderId: String, domain: String, resultCredential: @escaping (String) -> Void) {
+    sku.fetchOrderCredentials(domain, orderId: orderId) { completion in
+      Logger.module.debug("skus fetchOrderCredentials")
+      resultCredential(completion)
+    }
+  }
+  
+  func prepareCredentialsPresentation(for domain: String, path: String,
+                                      resultCredential: ((String) -> Void)?) {
+    Logger.module.debug("skus prepareCredentialsPresentation")
+    sku.prepareCredentialsPresentation(domain, path: path) { credential in
+      if !credential.isEmpty {
+        if let vpnCredential = BraveSkusWebHelper.fetchVPNCredential(credential, domain: domain) {
+          Preferences.VPN.skusCredential.value = credential
+          Preferences.VPN.skusCredentialDomain.value = domain
+          Preferences.VPN.skusCredentialExpirationDate.value = vpnCredential.expirationDate
+          
+          BraveVPN.setCustomVPNCredential(vpnCredential)
+        }
+      } else {
+        Logger.module.debug("skus empty credential from prepareCredentialsPresentation call")
+      }
+      
+      resultCredential?(credential)
+    }
+  }
+  
+  func credentialSummary(for domain: String, resultJSON: @escaping (Any?) -> Void) {
+    sku.credentialSummary(domain) { [weak self] completion in
+      do {
+        Logger.module.debug("skus credentialSummary")
+        
+        guard let data = completion.data(using: .utf8) else { return }
+        let json = try JSONSerialization.jsonObject(with: data, options: .fragmentsAllowed)
+        
+        resultJSON(json)
+        
+        let jsonDecoder = JSONDecoder()
+        jsonDecoder.keyDecodingStrategy = .convertFromSnakeCase
+        let credentialSummaryJson = try jsonDecoder.decode(CredentialSummary.self, from: data)
+        
+        if credentialSummaryJson.expirationDate == nil {
+          assertionFailure("Failed to parse expiration date")
+        }
+        
+        if credentialSummaryJson.isValid, let expirationDate = credentialSummaryJson.expirationDate {
+          Preferences.VPN.expirationDate.value = expirationDate
+          
+          // The credential has not expired yet, we can proceed with preparing it.
+          if expirationDate > Date() {
+            self?.prepareCredentialsPresentation(for: domain, path: "*", resultCredential: nil)
```

Comment:
We only call `prepareCredentialsPresentation` if the credential summary is valid and if not expired

Reply:
We don't need to check date on `credentialSummary` response - because it's when the monthly subscription ends (and not when credential expires). But the other checks you have are perfect 🙂 (active = true; count > 0)

Reply:
removed the date check